R/BackendHdf5-class.R

```diff
@@ -0,0 +1,296 @@
+#' @include hidden_aliases.R
+NULL
+
+#' Backend class that stores data temporarily to hdf5 files.
+#'
+#' @author Johannes Rainer, Sebastian Gibb
+#'
+#' @md
+#'
+#' @noRd
+setClass("BackendHdf5", contains = "Backend", slots = c(checksums = "character",
+                                                        h5files = "character"))
+
+#' @rdname hidden_aliases
+setMethod("show", "BackendHdf5", function(object) {
+    callNextMethod()
+    cat("Hdf5 path:", unique(dirname(object@h5files)), "\n")
+})
+
+#' @rdname Backend
+BackendHdf5 <- function() new("BackendHdf5")
+
+.valid.BackendHdf5.checksums <- function(x, y) {
+    if (length(x) != length(y))
+        "different number of md5 sums and hdf5 files"
+    else NULL
+}
+
+.valid.BackendHdf5.h5files <- function(x, y) {
+    msg <- NULL
+    if (length(x) != length(y))
+        msg <- "different number of hdf5 and original files"
+    if (anyDuplicated(x))
+        msg <- c(msg, "hdf5 files are not unique")
+    if (length(msg)) msg
+    else NULL
+}
+
+.valid.BackendHdf5.h5files.exist <- function(x) {
+    if (!all(file.exists(x)))
+        paste0("file(s) ", paste0(x[!file.exists(x)]), " not found")
+    else NULL
+}
+
+setValidity("BackendHdf5", function(object) {
+    msg <- c(.valid.BackendHdf5.checksums(object@files, object@checksums),
+             .valid.BackendHdf5.h5files(object@h5files, object@files),
+             .valid.BackendHdf5.h5files.exist(object@h5files))
+    if (length(msg)) msg
+    else TRUE
+})
+
+#' Initialize the Hdf5 backend, i.e. create the hdf5 files and add their names
+#' to the @h5files slot.
+#'
+#' @param object `BackendHdf5`
+#'
+#' @param files `character` with the original file names.
+#'
+#' @param spectraData `DataFrame` with the spectrum metadata.
+#'
+#' @param path `character(1)` defining the path where the hdf5 files should be
+#'     stored.
+#'
+#' @return `BackendHdf5` object.
+#'
+#' @author Johannes Rainer
+#'
+#' @md
+#'
+#' @noRd
+setMethod("backendInitialize", "BackendHdf5", function(object, files,
+                                                       spectraData, path = ".",
+                                                       ...) {
+    path <- normalizePath(path, mustWork = FALSE)
+    dir.create(path, showWarnings = FALSE, recursive = TRUE)
+    n_files <- length(files)
+    object@files <- files
+    names(object@files) <- sprintf(
```

Comment:
I wanted the names for the `file` slot for easier subsetting in the `BackendMemory`. Is it now useful at all? If not you can remove it here (and I will removed it from `Backend` and `BackendMemory`).

Reply:
Well, it is not used anymore, so we can remove it I guess.

Reply:
I've removed the code also from the `Backend` and `BackendMemory` classes to fix the unit tests. You should have a look after merging if it's OK and do some cleanup (I mostly just commented out the stuff but didn't delete).